gprofiler/profilers/profiler_base.py

```diff
@@ -75,32 +76,28 @@ def __init__(
         self,
         frequency: int,
         duration: int,
-        stop_event: Optional[Event],
-        storage_dir: str,
-        insert_dso_name: bool,
-        profiling_mode: str,
+        profiler_state: ProfilerState,
     ):
         self._frequency = limit_frequency(
-            self.MAX_FREQUENCY, frequency, self.__class__.__name__, logger, profiling_mode
+            self.MAX_FREQUENCY, frequency, self.__class__.__name__, logger, profiler_state.profiling_mode
         )
         if self.MIN_DURATION is not None and duration < self.MIN_DURATION:
             raise ValueError(
                 f"Minimum duration for {self.__class__.__name__} is {self.MIN_DURATION} (given {duration}), "
                 "raise the duration in order to use this profiler"
             )
         self._duration = duration
-        self._stop_event = stop_event or Event()
-        self._storage_dir = storage_dir
-        self._profiling_mode = profiling_mode
+        self._profiler_state = profiler_state
+        self._stop_event = self._profiler_state.stop_event or Event()
```

Comment:
Btw - is the `or Event()` actually needed? I think it's legacy code. You can try removing it in one commit and seeing if tests pass. If so, I'm in favor of removing it.